src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -47,9 +47,9 @@
     [
         RegexLexer(
             "inline_comment",
-            r"(-- |#)[^\n]*",
+            r"(^--|-- |#)[^\n]*",
```

Comment:
Does the non-spaced version need to be at the start of line? Would this be valid?
```
SELECT foo; --Comment
```

Reply:
I suppose that this would need testing as the documentation is indirectly suggesting otherwise and practice shows the '--' at the start of the line not followed with a space is taken as a comment.

MySql writes that they need a space ater the '--' contrary to standard sql, but at the same time in practice the sql files that I am setting up sqlfluf for do not have this space after the '--' that are at the start of the line and there is no issue reading these files.

Personnaly I would limit this to the start of the line given that elsewhere this may result in unwanted behavior.

https://dev.mysql.com/doc/refman/8.0/en/ansi-diff-comments.html

Reply:
Yeah, I agree with your reading of those docs, cheers 👍